### DISCUSSION TITLE
Is it possible to configure for this variant?

### DISCUSSION THREAD

**Discussion by TurtleKitty** [fairy-stockfish/Fairy-Stockfish#716] [Variants]
It's called Kozune: https://www.chessvariants.com/rules/kozune

There's nothing too strange about the board or pieces (basically compounds of WFADN), but there are a couple of rules I can't figure out how to do in variants.ini:

1) Pawns can choose whether to promote, like in Shogi. Pawns promote to weak pieces on rank 7, stronger ones on rank 8, and the strongest on rank 9.

2) The king can spawn a new pawn in an adjacent square if that file has no pawn of his color.

I guess while I'm here, I'll ask about a third thing unrelated to the above game: is it possible to start with pieces or pawns in-hand for games with drops? I couldn't figure that one out either.

**Comment by ianfab** [comment]
1. Different promotions based on the rank are not possible.
2. This mechanism is not supported.
3. Pieces in hand are put between the square brackets in the FEN, e.g., like in https://github.com/fairy-stockfish/Fairy-Stockfish/blob/e408eb87cfcb18529603e22ec8c59caf8183a3dd/src/variants.ini#L1008

**Reply by TurtleKitty** [reply]
Thanks! Good to know.
